File: theremin_backend/src/models/Scale.ts
import mongoose from 'mongoose';
const { Schema, model } = mongoose;

const scaleSchema = new Schema({
    type: { type: String, enum: ['Major', 'Minor', 'Major Pentatonic', 'Minor Pentatonic', 'Dorian', 'Lydian', 'Mixolydian', 'Phrygian', 'Harmonic Minor', 'Melodic Minor', 'Blues', 'Chromatic'] },
    rootNote: Number,
    intervals: [Number]
}, {
    timestamps: true
});

const Scale = model('Scale', scaleSchema);
export default Scale;

File: theremin_mar24/Assets/Scale.cs
using System;

[Serializable]
public class Scale
{
    public string scaleName;
    public int rootNote;
    public int[] intervals;

    public Scale(string name, int root, int[] intervals)
    {
        scaleName = name;
        rootNote = (root >= 0 && root <= 11) ? root : 0;
        this.intervals = intervals;
    }
}